packages/core/src/providers/oauth.ts

```diff
@@ -204,6 +204,14 @@ export interface OAuth2Config<Profile>
    * See [`oauth4webapi` client](https://github.com/panva/oauth4webapi/blob/main/docs/interfaces/Client.md) for details.
    */
   client?: Partial<Client>
+
+  /*
+   * Additional options for the underlying OAuth library
+   * `clientPrivateKey` is required if client.token_endpoint_auth_method === "private_key_jwt"
+   */
+  clientOptions?: {
+    clientPrivateKey?: string
+  }
```

Comment:
I think we should add `clientPrivateKey` on the `token` property instead, and it should accept `CryptoKey` as type, passing it directly to `oauth4webapi`

Reply:
I've made the change as you suggested, thanks for the pointer. As a result I've also had to update the `normalizeEndpoint` method to ensure `clientPrivateKey` is not stripped out before the callback. Let me know if that's OK or needs changing 👍

https://github.com/leemcmullen/next-auth/blob/21e2aa899541d6d815279d311032a801f404b4a9/packages/core/src/lib/utils/providers.ts#L158